File: app-presentation/src/main/java/com/zxjdev/smile/presentation/moment/list/di/MomentListFragmentModule.java
package com.zxjdev.smile.presentation.moment.list.di;

import com.zxjdev.smile.presentation.communal.di.scope.FragmentScope;
import com.zxjdev.smile.presentation.moment.list.MomentListContract;
import com.zxjdev.smile.presentation.moment.list.MomentListPresenter;

import dagger.Module;
import dagger.Provides;

@Module
public class MomentListFragmentModule {

  private MomentListContract.View view;

  public MomentListFragmentModule(MomentListContract.View view) {
    this.view = view;
  }

  @Provides
  @FragmentScope
  MomentListContract.View provideView() {
    return view;
  }

  @Provides
  @FragmentScope
  MomentListContract.Presenter providePresenter(MomentListPresenter presenter) {
    return presenter;
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/moment/list/MomentListFragmentModule.kt
package com.zxjdev.smile.presentation.moment.list

import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.ViewModelProviders
import com.zxjdev.smile.presentation.common.di.scope.FragmentScope
import dagger.Module
import dagger.Provides

@Module
class MomentListFragmentModule {

    @Provides
    fun provideMomentListViewModel(fragment: MomentListFragment): MomentListViewModel {
        return ViewModelProviders.of(fragment.requireActivity()).get(MomentListViewModel::class.java)
    }

    @Provides
    @FragmentScope
    fun provideLifecycleOwner(fragment: MomentListFragment): LifecycleOwner {
        return fragment.requireActivity()
    }
}